Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -3912,17 +3912,70 @@ void Parser::check_identifier_name_for_assignment_validity(StringView name, bool
     }
 }
 
+bool Parser::match_assert_clause() const
+{
+    return !m_state.current_token.trivia_contains_line_terminator() && m_state.current_token.original_value() == "assert"sv;
+}
+
+// AssertClause, https://tc39.es/proposal-import-assertions/#prod-AssertClause
+void Parser::parse_assert_clause(ModuleRequest& request)
+{
+    VERIFY(m_state.current_token.original_value() == "assert"sv);
+    consume(TokenType::Identifier);
+    consume(TokenType::CurlyOpen);
+
+    while (!done() && !match(TokenType::CurlyClose)) {
+        String key;
+        if (match(TokenType::StringLiteral)) {
+            key = parse_string_literal(m_state.current_token)->value().to_string();
+            consume();
+        } else if (match_identifier_name()) {
+            key = consume().value();
+        } else {
+            expected("IdentifierName or StringValue as AssertionKey");
+            consume();
+        }
+
+        consume(TokenType::Colon);
+
+        if (match(TokenType::StringLiteral)) {
+            for (auto& entries : request.assertions) {
+                if (entries.key == key)
+                    syntax_error(String::formatted("Duplicate assertion clauses with name: {}", key));
+            }
+            request.add_assertion(move(key), parse_string_literal(m_state.current_token)->value().to_string());
+        }
+        consume(TokenType::StringLiteral);
+
+        if (match(TokenType::Comma))
+            consume(TokenType::Comma);
+        else
+            break;
+    }
+
+    consume(TokenType::CurlyClose);
+}
+
 NonnullRefPtr<ImportStatement> Parser::parse_import_statement(Program& program)
 {
+    // We use the extended syntax which adds:
+    //  ImportDeclaration:
+    //      import ImportClauseFromClause [no LineTerminator here] AssertClause;
```

Comment:
needs one more space :^)

```suggestion
    //      import ImportClause FromClause [no LineTerminator here] AssertClause;
```